preliz/ppls/pymc_io.py

```diff
@@ -201,11 +200,55 @@ def reshape_params(model, var_info, p_model, params):
     return value
 
 
-def non_constant_parents(var_):
+def non_constant_parents(var_, free_rvs):
     """Find the parents of a variable that are not constant."""
     parents = []
     for variable in var_.get_parents()[0].inputs[2:]:
         if not isinstance(variable, TensorConstant):
-            parents.append(variable.owner.inputs[0])
-
+            for free_rv in free_rvs:
+                if free_rv in list(ancestors([variable])) and free_rv not in parents:
+                    parents.append(free_rv)
     return parents
+
+
+def posterior_to_prior(model, posterior, alternative=None):
+    """
+    Updates the priors of a probabilistic model by fitting them to posterior data, using either predefined or
+    user-specified alternative distributions. It selects the best-fitting distribution for each variable based
+    on maximum likelihood estimation (MLE). The result is a model with priors better aligned to the observed data.
+
+    Parameters
+    ----------
+    model : A PyMC model
+    A probabilistic model
+
+    posterior : Posterior samples
+    InferenceData from with the posterior group
+
+    alternative : "auto", list, dict, defaults to None
+    Users can add the model variables to consider alternative distributions while fitting samples
+
+    """
+
+    model_info = get_model_information(model)[2]
+    parsed_info = [(dist, var) for var, dist in model_info.items()]
+    new_priors = []
+
+    for dist, var in parsed_info:
+        dists = [model_info[var]]
+
+        if alternative == "auto":
+            dists += [Normal(), HalfNormal(), Gamma()]
+        elif isinstance(alternative, list):
+            dists += alternative
+        elif isinstance(alternative, dict):
+            dists += alternative.get(var, [])
+        if len(dists) == 1:
```

Comment:
Be explicit, one branch if alternative is None and another otherwise. inside the second branch evaluate the 3 other possible values for alternative.